ui/pages/confirmations/hooks/useAutomaticGasFeeTokenSelect.ts

```diff
@@ -0,0 +1,48 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useConfirmContext } from '../context/confirm';
+import { useAsyncResult } from '../../../hooks/useAsync';
+import { updateSelectedGasFeeToken } from '../../../store/controller-actions/transaction-controller';
+import { forceUpdateMetamaskState } from '../../../store/actions';
+import { getIsSmartTransaction } from '../../../../shared/modules/selectors';
+import { useInsufficientBalanceAlerts } from './alerts/transactions/useInsufficientBalanceAlerts';
+
+export function useAutomaticGasFeeTokenSelect() {
+  const dispatch = useDispatch();
+  const isSmartTransaction = useSelector(getIsSmartTransaction);
+
+  const { currentConfirmation: transactionMeta } =
+    useConfirmContext<TransactionMeta>();
+
+  const hasInsufficientBalance = Boolean(
+    useInsufficientBalanceAlerts()?.length,
+  );
+
+  const {
+    gasFeeTokens,
+    id: transactionId,
+    selectedGasFeeToken,
+  } = transactionMeta;
+
+  const firstGasFeeTokenAddress = gasFeeTokens?.[0]?.tokenAddress;
+
+  const selectFirstToken = useCallback(async () => {
+    await updateSelectedGasFeeToken(transactionId, firstGasFeeTokenAddress);
+    await forceUpdateMetamaskState(dispatch);
+  }, [dispatch, transactionId, firstGasFeeTokenAddress]);
+
+  const shouldSelect =
+    isSmartTransaction &&
+    hasInsufficientBalance &&
```

Comment:
Q: why is the insufficient balance a required check for updating the selected gas fee token? 

option to add a test for this as well

Reply:
This logic is specifically for automatically selecting the first token, so we only want to do that if the user can't afford to pay natively.

Thought I'd already added that unit test, will add it now.

Reply:
Adding this unit test in the follow up metrics PR.